Source/Events.Processing/Streams/StreamProcessorState.cs

```diff
@@ -23,6 +23,22 @@ public StreamProcessorState(StreamPosition streamPosition)
             RetryTime = DateTimeOffset.UtcNow;
             FailureReason = string.Empty;
             ProcessingAttempts = 0;
```

Comment:
Same comment as above with the constructor that does not make sense to expose publically since it should only be used once. Make it either private or have private setters on the properties